yotiprofile.go

```diff
@@ -0,0 +1,146 @@
+package yoti
+
+import (
+	"github.com/getyoti/yoti-go-sdk/attribute"
+)
+
+const (
+	attrConstSelfie                  = "selfie"
+	attrConstGivenNames              = "given_names"
+	attrConstFamilyName              = "family_name"
+	attrConstFullName                = "full_name"
+	attrConstMobileNumber            = "phone_number"
+	attrConstEmailAddress            = "email_address"
+	attrConstDateOfBirth             = "date_of_birth"
+	attrConstAddress                 = "postal_address"
+	attrConstStructuredPostalAddress = "structured_postal_address"
+	attrConstGender                  = "gender"
+	attrConstNationality             = "nationality"
+)
+
+//Profile represents the details retrieved for a particular
+type Profile struct {
+	// AttributeSlice represents a map of the Yoti attributes, each attribute is a small piece of information about a Yoti user such as a photo of the user or the
+	// user's date of birth.
+	AttributeSlice []*attribute.Attribute
+}
+
+// Selfie is a photograph of the user. Will be nil if not provided by Yoti
+func (p Profile) Selfie() *attribute.ImageAttribute {
+	for _, a := range p.AttributeSlice {
+		if a.Name == attrConstSelfie {
+			return attribute.NewImageAttribute(a)
+		}
+	}
+	return nil
+}
+
+// GivenNames represents the user's given names. Will be nil if not provided by Yoti
+func (p Profile) GivenNames() *attribute.StringAttribute {
+	for _, a := range p.AttributeSlice {
+		if a.Name == attrConstGivenNames {
+			return attribute.NewStringAttribute(a)
+		}
+	}
+	return nil
+}
```

Comment:
This seems to imply that a profile has only a single given name. This may not be true as a user can register more than a single document each possibly resulting in more than a single given_names attribtues

Reply:
None of the SDKs are set up to have multiple attributes with the same name. We can discuss this with Quirino.

Reply:
Yes, I think this is worth discussing. It maybe that they have done something similar to the client, which is to have a priority order here and choose the top priority to return... we would still support multiple right up to this point I think.

Reply:
This is something which is likely to be implemented in the SDKs in the future. Since it will require quite a large overhaul for each one, we won't be doing it right now.